tests/units/test_runner.py

```diff
@@ -182,19 +165,27 @@ async def test_prepare_tests_with_specific_tests(caplog: pytest.LogCaptureFixtur
     assert sum(len(tests) for tests in selected_tests.values()) == 5
 
 
-async def test_runner_dry_run(caplog: pytest.LogCaptureFixture, test_inventory: AntaInventory) -> None:
-    """Test that when dry_run is True, no tests are run.
-
-    caplog is the pytest fixture to capture logs
-    test_inventory is a fixture that gives a default inventory for tests
-    """
-    logger.setup_logging(logger.Log.INFO)
+async def test_dry_run(caplog: pytest.LogCaptureFixture, inventory: AntaInventory) -> None:
+    """Test that when dry_run is True, no tests are run."""
     caplog.set_level(logging.INFO)
     manager = ResultManager()
-    catalog_path = Path(__file__).parent.parent / "data" / "test_catalog.yml"
-    catalog = AntaCatalog.parse(catalog_path)
+    await main(manager, inventory, FAKE_CATALOG, dry_run=True)
+    assert "Dry-run mode, exiting before running the tests." in caplog.records[-1].message
 
-    await main(manager, test_inventory, catalog, dry_run=True)
 
-    # Check that the last log contains Dry-run
-    assert "Dry-run" in caplog.records[-1].message
+async def test_cannot_create_test(caplog: pytest.LogCaptureFixture, inventory: AntaInventory) -> None:
+    """Test that when an Exception is raised during test instantiation, it is caught and a log is raised."""
+    caplog.set_level(logging.CRITICAL)
+    manager = ResultManager()
+    catalog = AntaCatalog.from_list([(FakeTestWithMissingTest, None)])  # type: ignore[type-abstract]
+    await main(manager, inventory, catalog)
+    msg = (
+        "There is an error when creating test tests.units.test_models.FakeTestWithMissingTest.\nIf this is not a custom test implementation: "
+        "Please reach out to the maintainer team or open an issue on Github: https://github.com/aristanetworks/anta.\nTypeError: "
+    )
+    msg += (
+        "Can't instantiate abstract class FakeTestWithMissingTest without an implementation for abstract method 'test'"
+        if sys.version_info >= (3, 12)
+        else "Can't instantiate abstract class FakeTestWithMissingTest with abstract method test"
```

Comment:
maybe lets make some helped somewhere?

Reply:
I don't think that would improve readability of unit tests, and it's only 2 of them